### PR TITLE
Cleaner integration tests

### PR DESCRIPTION
**Pull request by wblachowski**
Added cleaner integration tests as suggested in #441

## Description
My solution uses a [mocked osgi context](https://sling.apache.org/documentation/development/osgi-mock.html) as well as [in-memory mock of mongoDB server](https://github.com/bwaldvogel/mongo-java-server). It checks various combinations of cleaner params (versions to keep and suite max age) and verifies whether correct metadata and artifacts documents have been removed from database.

For testing data I used real json objects extracted from my local aet instance db after a couple of runs (with some timestamp changes).

To make testing possible I had to incorporate some changes in cleaner code. This includes `LocalDateTimeProvider` interface to mock current system time and `CamelContextCreator` to inject different camel context configuration during testing and normal execution.

## Motivation and Context
Refers to enhancement brought up in #441

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [x] I have reviewed (and updated if needed) the documentation regarding this change

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

**Comment by plutasnyy** [comment]
Good job!

File: core/cleaner/pom.xml [hunk 1]
```diff
@@ -33,6 +33,22 @@
   <name>AET :: Core :: Cleaner</name>
   <description>Cleaner responsible for removing outdated AET artifacts from datastorage</description>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
```
Comment (tkaik) [inline core/cleaner/pom.xml]: If those are dependencies only for tests then we should probably add`<scope>test</scope>`

Comment (wblachowski) [reply core/cleaner/pom.xml]: done :wink:

File: integration-tests/cleaner-test/pom.xml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+  <parent>
+    <artifactId>aet-root</artifactId>
+    <groupId>com.cognifide.aet</groupId>
+    <version>3.2.3-SNAPSHOT</version>
+    <relativePath>../..</relativePath>
+  </parent>
+
+  <artifactId>cleaner-test</artifactId>
+  <packaging>jar</packaging>
```
Comment (Skejven) [inline integration-tests/cleaner-test/pom.xml]: Wy do we need `jar` distribution of this module?
It is only for tests.

File: integration-tests/cleaner-test/src/main/java/com/cognifide/aet/cleaner/InMemoryDB.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import com.mongodb.client.MongoDatabase;
+import de.bwaldvogel.mongo.MongoServer;
+import de.bwaldvogel.mongo.backend.memory.MemoryBackend;
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import org.apache.commons.io.FileUtils;
+import org.bson.Document;
+
+public class InMemoryDB {
```
Comment (Skejven) [inline integration-tests/cleaner-test/src/main/java/com/cognifide/aet/cleaner/InMemoryDB.java]: Shouldn't all files in this module be in `src/test`?
Why do we need any of the code to be non-test?

File: integration-tests/cleaner-test/pom.xml (hunk 2 above)
Comment (tkaik) [reply integration-tests/cleaner-test/pom.xml]: Yes, if we set it to `pom` then the tests are not being compiled or executed

File: integration-tests/cleaner-test/src/main/java/com/cognifide/aet/cleaner/InMemoryDB.java (hunk 3 above)
Comment (tkaik) [reply integration-tests/cleaner-test/src/main/java/com/cognifide/aet/cleaner/InMemoryDB.java]: You're right, I moved all files to `src/test`

**Pull request merged**